### ISSUE TITLE
GridFS WritableStream throws exception with recent driver

### ISSUE THREAD

**Issue by doublecompile** [mongodb/mongo-php-library#251]
I'm running HHVM 3.15 using the latest dev master HHVM Driver. Since [HHVM-162](https://jira.mongodb.org/browse/HHVM-162) was implemented, the `MongoDB\GridFS\WritableStream` constructor is throwing a `MongoDB\Driver\Exception\InvalidArgumentException` exception because it's passing a float as a constructor argument to `UTCDateTime`.

```
Error parsing "1474912381598" as 64-bit integer for MongoDB\BSON\UTCDateTime initialization
#0 /usr/share/nginx/vendor/mongodb/mongodb/src/GridFS/WritableStream.php(88): MongoDB\BSON\UTCDateTime->__construct()
#1 /usr/share/nginx/vendor/mongodb/mongodb/src/GridFS/StreamWrapper.php(228): MongoDB\GridFS\WritableStream->__construct()
#2 /usr/share/nginx/vendor/mongodb/mongodb/src/GridFS/StreamWrapper.php(93): MongoDB\GridFS\StreamWrapper->initWritableStream()
#3 (): MongoDB\GridFS\StreamWrapper->stream_open()
#4 /usr/share/nginx/vendor/mongodb/mongodb/src/GridFS/Bucket.php(359): fopen()
#5 /usr/share/nginx/vendor/mongodb/mongodb/src/GridFS/Bucket.php(416): MongoDB\GridFS\Bucket->openUploadStream()
```

The line in question is:

``` php
            'uploadDate' => new UTCDateTime(floor(microtime(true) * 1000)),
```

Which, of course, bombs out because `UTCDateTime` doesn't accept `float` values any longer.

**Comment by moderndeveloperllc** [comment]
There are quite a few things that will need to be updated once 1.2.0 lands in both the library and documentation. @jmikola Perhaps a 1.2.0 branch on this repo would be appropriate so that alpha/beta testers can use the library while testing?

**Comment by jmikola** [comment]
Cross-referencing with https://github.com/mongodb/mongo-php-driver/issues/426 (even though this pertains to HHVM).

@derickr: I believe this is due to https://github.com/mongodb/mongo-hhvm-driver/commit/ca19876ca97a1d916441f4435cb7e3d9612bc1e2 removing the default `(string)` cast. I can fix this quickly in PHPLIB but I'd consider resolving this upstream before 1.2.0 (akin to [PHPC-804](https://jira.mongodb.org/browse/PHPC-804)) since `floor(microtime(true) * 1000)` was quite common for code using the 1.1.x extension API.

**Comment by derickr** [comment]
Passing in a float, will however, cause potential data loss without knowing about it. UTCDateTime only supports integers, so I consider this correct behaviour. The original allowing of float, was IMO a bug.

**Comment by jmikola** [comment]
Tracking ticket: [PHPLIB-222](https://jira.mongodb.org/browse/PHPLIB-222).

> Perhaps a 1.2.0 branch on this repo would be appropriate so that alpha/beta testers can use the library while testing?

@moderndeveloperllc: I'm in the process of tagging a 1.1.0 alpha release today to get the GridFS code out there. The `master` branch is currently `1.1.x-dev` and 1.1.0 will depend on the extensions' 1.2.0 API. I'm not sure I understand how a compatibility branch for `ext-mongodb` 1.2.0 would help (if that's what you were suggesting).

> The original allowing of float, was IMO a bug.

@derickr: Noted. I'll be fixing this in the library but keep this in mind in case you get future bug reports once folks are testing the 1.2.0 HHVM extension.

**Comment by jmikola** [comment]
@moderndeveloperllc: I got caught up with https://github.com/mongodb/mongo-php-library/pull/254 this afternoon, but I believe that resolves all of the current HHVM issues in `master`. Hoping to proceed with the alpha release tomorrow morning.

**Comment by jmikola** [comment]
I realize I never followed up after [1.1.0-alpha1](https://github.com/mongodb/mongo-php-library/releases/tag/1.1.0-alpha1) was released.

Additionally, we just released [1.2.0](https://github.com/mongodb/mongo-php-driver/releases/tag/1.2.0) of the extension, which includes [PHPC-790](https://jira.mongodb.org/browse/PHPC-790) to ensure that floats passed to the UTCDateTime constructor are truncated. [PHPC-804](https://jira.mongodb.org/browse/PHPC-804) also ensures we don't yield any errors with PHP 7's strict types mode.

Closing this out, as everything should be resolved.
